sky/resources.py

```diff
@@ -275,7 +275,35 @@ def _set_region_zone(self, region: Optional[str],
         self._region, self._zone = self._cloud.validate_region_zone(
             region, zone)
 
-    def _try_validate_instance_type(self):
+    def get_valid_region_zones(self) -> List[Tuple[clouds.Region, clouds.Zone]]:
+        """Returns a list of (region, zone) that can provision this Resources"""
+        assert self.is_launchable()
+
+        if isinstance(self._cloud, clouds.GCP):
+            gcp_region_zones = list(
+                self._cloud.region_zones_provision_loop(
+                    instance_type=self._instance_type,
+                    accelerators=self.accelerators,
+                    use_spot=self._use_spot))
+
+            # GCP provision loop yields 1 zone per request.
+            # For consistency with other clouds, we group the zones in the
+            # same region.
+            region_zones = []
+            regions = set()
+            for region, _ in gcp_region_zones:
+                if region.name not in regions:
+                    regions.add(region.name)
+                    region_zones.append((region, region.zones))
```

Comment:
Document something like: `# This utilizes the knowledge that GCP.region_zones_provision_loop() will set the zones field of each returned Region to be the resource-offering zones.` (Knowledge leakage, but I'm not sure how to get rid of it; maybe it's ok for now.)

Reply:
Added. Thanks!